docs/overlay.md

```diff
@@ -1,6 +1,146 @@
-Overlay Driver
```

Comment:
This document is supposed to carry the design details for overlay driver.
I prefer if you can add a new file named overlay-quckstart-config.md or something.

If you prefer to keep it in 1 file, Atleast I will keep the heading on `Overlay Driver` and also add a section for `Design` and mark it as TODO, followed by the quick start guide.